status/reporters.go

```diff
@@ -480,6 +488,14 @@ func (r *GitHubReporter) ReportStatusForSnapshot(k8sClient client.Client, ctx co
 
 		for _, integrationTestStatusDetail := range integrationTestStatusDetails {
 			integrationTestStatusDetail := *integrationTestStatusDetail // G601
+			if latestUpdateTime.Before(integrationTestStatusDetail.LastUpdateTime) {
+				logger.Info("IntegrationTest: %w contains new changes", integrationTestStatusDetail.ScenarioName)
+				if newLatestUpdateTime.Before(integrationTestStatusDetail.LastUpdateTime) {
+					newLatestUpdateTime = integrationTestStatusDetail.LastUpdateTime
+				}
+			} else {
+				logger.Info("IntegrationTest: %w contains no new changes", integrationTestStatusDetail.ScenarioName)
```

Comment:
Should we call a `continue` after this or something similar in order to skip executing creating/updating a checkRun?

Reply:
Thats what I had there in a first place, it invalid memory address or nil pointer dereference in unit tests. I thought that changes in https://github.com/redhat-appstudio/integration-service/pull/437 should take care of that.

Reply:
if we don't put `continue` here, we don't even need this PR

Reply:
Also we shouldn't log this, it will flood logs with the same msg